File: youreffect/src/main/java/com/youreffect/model/Item.java
package com.youreffect.model;


/**
 * @author Deeban Ramalingam
 * Item POJO
 */
public class Item {

    private String itemId;
    private String userId;
    private String name;
    private String energy;
    private int quantity;
    private String specs;
    private String dateCreated;

    public String getItemId() {
        return itemId;
    }

    public void setItemId(String itemId) {
        this.itemId = itemId;
    }

    public String getUserId() {
        return userId;
    }

    public void setUserId(String userId) {
        this.userId = userId;
    }

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public String getEnergy() {
        return energy;
    }

    public void setEnergy(String energy) {
        this.energy = energy;
    }

    public int getQuantity () {
        return quantity;
    }

    public void setQuantity (int quantity) {
        this.quantity = quantity;
    }

    public String getSpecs() {
        return specs;
    }

    public void setSpecs(String specs) {
        this.specs = specs;
    }

    public String toString() {
        return String.format("Item[id=%s, userId='%s', name='%s', energy='%s', quantity='%s', dateCreated='%s', specs='%s']", itemId, userId, name, energy, quantity, dateCreated, specs);
    }

    public String getDateCreated() {
        return dateCreated;
    }

    public void setDateCreated(String dateCreated) {
        this.dateCreated = dateCreated;
    }
}

File: youreffect/src/main/webapp/public/assets/js/model/item.js
/**
 * item front-end model
 * @param name
 * @param energy
 * @constructor construct item with name and energy form
 */
function Item(name, energy) {
    this.itemId = '';
    this.userId = '';
    this.name = name;
    this.energy = energy;
    this.quantity = 0;
    this.specs = new Object();
}

/**
 * adds key/value spec pair to specs map of item
 * @param key mapped key of spec
 * @param value mapped value of spec
 */
Item.prototype.addSpec = function(key, value) {
    this.specs[key] = value;
};
